File: packages/core/src/__tests__/api-step.ts
import { ApiRouteHandler, FlowContext } from '../types'

export const handler: ApiRouteHandler = async (_, ctx: FlowContext) => {
  await ctx.emit({ data: { test: 'data' }, type: 'TEST_EVENT' })

  return { status: 200, body: { traceId: ctx.traceId } }
}

File: packages/core/src/__tests__/steps/api-step.py
from typing import Dict

config = {
    "type": "api",
    "name": "api-step", 
    "emits": ["TEST_EVENT"],
    "path": "/test",
    "method": "POST"
}


async def handler(_, context) -> Dict:
    await context.emit({
        "data": {"test": "data"},
        "topic": "TEST_EVENT"
    })
    
    return {
        "status": 200,
        "body": {"traceId": context.trace_id}
    }
